### ISSUE TITLE
Split testHowFarAheadOrBehindSchedule() into smaller test methods

### ISSUE THREAD

**Issue by knjk04** [Project-Books/book-project#393]
In `CalculateReadingGoalTest`, we currently have the following implementation for `testHowFarAheadOrBehindSchedule()`:

```java
    @Test
    void testHowFarAheadOrBehindSchedule(){
        Mockito.mockStatic(DateUtils.class);

        Mockito.when(DateUtils.getCurrentWeekNumberOfYear()).thenReturn(1);
        assertThat(howFarAheadOrBehindSchedule(52, 1)).isZero();
        assertThat(howFarAheadOrBehindSchedule(52,0)).isOne();
        assertEquals(9, howFarAheadOrBehindSchedule(52,10));
        assertEquals(9, howFarAheadOrBehindSchedule(199,12));
        assertEquals(2, howFarAheadOrBehindSchedule(199,5));

        Mockito.when(DateUtils.getCurrentWeekNumberOfYear()).thenReturn(15);
        assertEquals(12, howFarAheadOrBehindSchedule(52,3));
        assertEquals(9, howFarAheadOrBehindSchedule(52,24));
        assertEquals(5, howFarAheadOrBehindSchedule(52,20));

        Mockito.when(DateUtils.getCurrentWeekNumberOfYear()).thenReturn(10);
        assertEquals(20, howFarAheadOrBehindSchedule(199,50));
        assertEquals(22, howFarAheadOrBehindSchedule(199,8));
        assertEquals(70, howFarAheadOrBehindSchedule(199,100));

        Mockito.when(DateUtils.getCurrentWeekNumberOfYear()).thenReturn(43);
        assertEquals(7, howFarAheadOrBehindSchedule(113,79));
        assertEquals(45, howFarAheadOrBehindSchedule(113,41));
        assertThat(howFarAheadOrBehindSchedule(113,86)).isZero();
    }
```

This method is doing too much. We should refactor this so that we are testing one normal (acceptable) case and two boundary cases

**Comment by mofiebiger** [comment]
Hi @knjk04, can I work on this issue? Thanks! 😊

**Comment by knjk04** [comment]
Hi @mofiebiger, I saw that you recently requested to be assigned to another issue. In line with our [contributing guidelines](https://github.com/Project-Books/book-project/blob/master/CONTRIBUTING.md), we only assign one issue at a time to new contributors.

If this still issue is still available after completing the other issue you requested to be assigned to, feel to request again!

**Comment by mofiebiger** [comment]
Hi @knjk04, would it be OK to work on this one instead? It's my first time contributing to an open source project and this issue feels a bit more doable for a first go.

**Comment by knjk04** [comment]
Sure, go for it. I'll assign you to the issue.

I know contributing to open source for the first time can be daunting. If you need a hand with anything, feel free to get in touch with us over our [Slack](https://codecov.io/gh/Project-Books/book-project) workspace

**Comment by mofiebiger** [comment]
Thank you so much 👩‍💻

**Comment by mofiebiger** [comment]
Hey @knjk04 👋. I may have found a bug and wanted to double check if this was expected behaviour. Here is a link to my [Draft PR](https://github.com/Project-Books/book-project/pull/419/files), which has one failing test. 

In the method `booksToReadFromStartOfYear()`, when the number of books is less than the number of weeks in the year _(52)_ there is an issue due to integer division. For example, when the number of books to read in a year is _40_ then this method returns _0_ and not _.77_. Would this be expected behaviour? 

If this is expected then I will finish up refactoring the tests and if not then I can open up a new issue!

**Comment by knjk04** [comment]
Hey @mofiebiger. 

Thanks for bringing this up! You're right that this a bug. I recently noticed this, too! I've asked Bharat to fix it (see https://github.com/Project-Books/book-project/blob/cc361d7f7d9353d8274207101ed3a143842d38d7/src/main/java/com/karankumar/bookproject/backend/goal/CalculateReadingGoal.java#L51) 

By making it a `double` instead of an `int`, we shouldn't have this bug.

**Comment by mofiebiger** [comment]
Great! Thank you @knjk04 for getting back to me so quickly! I will finish refactoring the tests once this PR is merged - https://github.com/Project-Books/book-project/pull/398
